broker/lib/fabrik/DirectorManager.js

```diff
@@ -177,7 +189,179 @@ class DirectorManager extends BaseManager {
     return this.director.getDeploymentIps(deploymentName);
   }
 
+  executePolicy(scheduled, action, deploymentName, dbUpdate, policyApplicable) {
+    const runOutput = {
+      'directorError': false,
+      'shouldRunNow': false
+    };
+    let targetDirectorConfig;
+    return this.director.getDirectorForOperation(action, deploymentName)
+      .then(directorConfig => {
+        targetDirectorConfig = directorConfig;
+        return this.director.getCurrentTasks(action, targetDirectorConfig);
+      })
+      .then(tasksCount => {
+        if (!policyApplicable) {
+          runOutput.shouldRunNow = true;
+          return Promise.resolve(runOutput);
+        }
+        if (dbUpdate) {
```

Comment:
I will make this change. On the other hand, this would mean that the any operation with this `_runImmediately` flag set would not go through the `executePolicy` function at all. So, technically, MongoDB updates will not be resilient against BOSH outages. I guess that's ok